arbnode/dataposter/dataposter_test.go

```diff
@@ -152,46 +153,36 @@ func TestMaxFeeCapFormulaCalculation(t *testing.T) {
 	}
 }
 
-type stubL1Client struct {
+type stubL1ClientInterface struct {
 	senderNonce        uint64
 	suggestedGasTipCap *big.Int
-
-	// Define most of the required methods that aren't used by feeAndTipCaps
-	backends.SimulatedBackend
-}
-
-func (c *stubL1Client) NonceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (uint64, error) {
-	return c.senderNonce, nil
-}
-
-func (c *stubL1Client) SuggestGasTipCap(ctx context.Context) (*big.Int, error) {
-	return c.suggestedGasTipCap, nil
-}
-
-// Not used but we need to define
-func (c *stubL1Client) BlockNumber(ctx context.Context) (uint64, error) {
-	return 0, nil
-}
-
-func (c *stubL1Client) CallContractAtHash(ctx context.Context, msg ethereum.CallMsg, blockHash common.Hash) ([]byte, error) {
-	return []byte{}, nil
 }
 
-func (c *stubL1Client) CodeAtHash(ctx context.Context, address common.Address, blockHash common.Hash) ([]byte, error) {
-	return []byte{}, nil
+func (c *stubL1ClientInterface) CallContext(ctx_in context.Context, result interface{}, method string, args ...interface{}) error {
+	switch method {
+	case "eth_getTransactionCount":
+		ptr, ok := result.(*hexutil.Uint64)
+		if !ok {
+			return errors.New("result is not a *hexutil.Uint64")
+		}
+		*ptr = hexutil.Uint64(c.senderNonce)
+	case "eth_maxPriorityFeePerGas":
+		ptr, ok := result.(*hexutil.Big)
+		if !ok {
+			return errors.New("result is not a *hexutil.Big")
+		}
+		*ptr = hexutil.Big(*c.suggestedGasTipCap)
+	}
+	return nil
```

Comment:
Can we return an error instead, something like `fmt.Errorf("called unexpected method: %v", method)`?

Reply:
I dont think we need to return an error in default case because that would be inaccurate as we essentially want a stub `rpc.ClientInterface` that should only ever have meaningful responses to two methods of the main/outer `ethclient.Client` struct and those are `NonceAt` and `SuggestGasTipCap`, we can just ignore the rest instead of erroring there. This is in sync with how it was handled before (declaring empty methods on `stubL1Client`)